CMakeLists.txt

```diff
@@ -78,7 +78,14 @@ Generating build files for OpenShot
 #### Enable C++11 (for JUCE)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
+
+#### Optional C++ restirctions
+option(USE_CXX_GNU_EXTENSIONS "Allow GNU C++ extensions" OFF)
```

Comment:
For the reasons I explained in my general comment, I wouldn't mention "GNU" in the variable name or anywhere. This will enable GNU extensions only when the build is being run with `g++`. If it's using any other compiler it'll enable _**those**_ extensions instead. (Unless you want to wrap an `if(CMAKE_COMPILER_IS_GNUCC)` around the `set()` call that turns on extensions.)

Also, nit: "restrictions" is misspelled.